doc/ipython-notebooks/gaussian_process/variational_classifier.ipynb

```diff
@@ -629,14 +733,380 @@
+      "We use the banana data set to demonstrate decision boundary of GPC. The data set can be found at <a href=\"http://sci2s.ugr.es/keel/dataset.php?cod=182\">here</a>. This is a binary classification problem. The goal of this example is to visually show the decision boundary of GPC models. The following is the description of the data set.\n",
+      "\n",
+      "An artificial data set where instances belongs to several clusters with a banana shape. There are two attributes At1 and At2 corresponding to the x and y axis, respectively. The class label (-1 and 1) represents one of the two banana shapes in the dataset."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "def train(inference, linesearch, likelihood, x_train, y_train, kernel_log_sigma, kernel_log_scale):\n",
+      "    \n",
+      "    \"\"\"\n",
+      "    This function is used to train a GP classifer given an inference method\n",
+      "        \n",
+      "    Parameters:\n",
+      "        inference - an inference methods used to train the classifer\n",
+      "        linesearch - a linesearch method used in the inference method \n",
+      "        likelihood - likelihood function to model data\n",
+      "        x_train, x_test - X for training and testing\n",
+      "        y_train - labels for training\n",
+      "        kernel_log_sigma, kernel_log_scale hyper-parameters of covariance function    \n",
+      "    Returns:\n",
+      "        trained GPC model, name of inference method\n",
+      "    \"\"\"\n",
+      "\n",
+      "    mean_func = ZeroMean()\n",
+      "    kernel_sigma = 2*exp(2*kernel_log_sigma);\n",
+      "    kernel_func = GaussianKernel(10, kernel_sigma)\n",
+      "\n",
+      "    #Y is a sample-by-1 vector\n",
+      "    labels_train = BinaryLabels(y_train)\n",
+      "    #X is a feature-by-sample matrix\n",
+      "    features_train=RealFeatures(x_train)\n",
+      "\n",
+      "    inf = inference(kernel_func, features_train, mean_func, labels_train, likelihood)\n",
+      "    inf.set_scale(exp(kernel_log_scale))\n",
+      "\n",
+      "    gp = GaussianProcessClassification(inf)\n",
+      "    gp.train()\n",
+      "\n",
+      "    return gp, inf.get_name()"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "def extract1(inf):\n",
```

Comment:
Since this is parsing code, it pollutes the notebook reading flow a bit. 
What about adding a parsed file to shogun-data instead, that you can just load straight away? Maybe seperate files for labels and covariates?